hnn_core/network.py

```diff
@@ -143,6 +279,13 @@ class Network(object):
         Firing threshold of all cells.
     delay : float
         Synaptic delay in ms.
+
+    Notes
+    ----
+    `net = default_network(params)` is the reccomended path for creating a
+    network. Instantiating the network as `net = Network(params)` will
```

Comment:
we should migrate the stuff that is specific to `params` specification to `default_network` slowly leaving a generalized `Network` class in place. That is to say, it should not take a `params` dictionary as input and should not have `add_drives_from_params`. Happy to delegate this to another PR though

Reply:
I agree this is a great direction to go. Let's iterate on generalising `Network` and removing  `add_drives_from_params` together with the `legacy`-flag.